src/panel-chat.tsx

```diff
@@ -112,16 +112,77 @@ class ChatTextEntry extends preact.Component<{room: PSRoom, onMessage: (msg: str
 		elem.focus();
 	};
 	keyPress = (e: KeyboardEvent) => {
+		const cmdKey = (((e.cmdKey || e.metaKey) ? 1 : 0) + (e.ctrlKey ? 1 : 0) === 1) && !e.altKey && !e.shiftKey;
 		let elem = e.currentTarget as HTMLTextAreaElement;
-		if (e.keyCode === 13 && !e.shiftKey) {
+		if (e.key === 'Enter' && !e.shiftKey) {
 			this.props.onMessage(elem.value);
 			elem.value = '';
 			this.update();
 			e.preventDefault();
 			e.stopImmediatePropagation();
-			return;
+		} else if (e.key === 'i' && cmdKey) {
+			if (ChatTextEntry.toggleFormatChar(elem, '_')) {
+				e.preventDefault();
+				e.stopImmediatePropagation();
+			}
+		} else if (e.key === 'b' && cmdKey) {
+			if (ChatTextEntry.toggleFormatChar(elem, '*')) {
+				e.preventDefault();
+				e.stopImmediatePropagation();
+			}
+		} else if (e.key === '`' && cmdKey) {
+			// FIXME: doesn't work :(
+			if (ChatTextEntry.toggleFormatChar(elem, '`')) {
+				e.preventDefault();
+				e.stopImmediatePropagation();
+			}
 		}
 	};
+	static toggleFormatChar(elem, formatChar) {
```

Comment:
I guess you haven't run TypeScript yet.

```suggestion
	static toggleFormatChar(elem: HTMLTextareaElement, formatChar: string) {
```